packages/compass-preferences-model/src/preferences-schema.tsx

```diff
@@ -208,15 +205,16 @@ export type DeriveValueFunction<T> = (
   /** Get a preference's value from the current set of preferences */
   getValue: <K extends keyof AllPreferences>(key: K) => AllPreferences[K],
   /** Get a preference's state from the current set of preferences */
-  getState: <K extends keyof AllPreferences>(key: K) => PreferenceState
+  getState: <K extends keyof AllPreferences>(key: K) => PreferenceState,
+  atlasCloudFeatureFlags: Partial<AtlasCloudFeatureFlags>
```

Comment:
Passed as an extra argument to `deriveValue` function to avoid exposing this in some other way through actual preferences definitions: we don't want these to leak somehow through `AllPreferences` types